ddclient

```diff
@@ -2298,57 +2264,78 @@ sub get_ip {
 }
 
 ######################################################################
-## is_ipv4() validates if string is valid IPv4 address and only
-## a valid IPv4 address (no preceding or trailing spaces/characters)
+## is_ipv4() validates if string is valid IPv4 address and only a
+## valid address with no preceding or trailing spaces/characters
+## and no embedded leading zeros.
 ######################################################################
 sub is_ipv4 {
     my ($value) = @_;
-    return (length($value // '') != 0) && ($value eq (ipv4_match($value) // ''));
+    return (length($value // '') != 0) && ($value eq (ipv4_extract($value) // ''));
 }
 
 ######################################################################
-## ipv4_match() extracts the first valid IPv4 address from given string
+## ipv4_extract() extracts the first valid IPv4 address from given string.
+## Accepts leading zeros in the address but removes them in returned value
 ######################################################################
-sub ipv4_match {
-    (shift // '') =~ /\b((?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?))\b/ai;
-    return $1;
+sub ipv4_extract{
+    (shift // '') =~ /\b((?:(?<octet>25[0-5]|2[0-4]\d|[01]?\d\d?)\.){3} ## first three bytes
+                         (?&octet))\b/xa;                               ## last one (without period)
+    my $ip = ($1 // '');
+    $ip =~ s/\b0+\B//g;                                                ## remove leading zeros
+    return $ip;
```

Comment:
This function should return either an IP address or `undef`, never an empty string.

```suggestion
    if ((shift // '') !~ /\b((?:(?<octet>25[0-5]|2[0-4]\d|[01]?\d\d?)\.){3}(?&octet))\b/a) {
        return undef;
    }
    (my $ip = $1) =~ s/\b0+\B//g;  # remove leading zeros
    return $ip;
```

Reply:
Thanks for the suggestion, change made.